plugins.md

```diff
@@ -0,0 +1,218 @@
+# Trace Plugin API Developer Guide
+
+**Note: The Trace Plugin API is experimental and is subject to change without notice.**
+
+The trace agent is driven by a set of plugins that describe how to patch a
+module to generate trace spans when that module is used. We provide plugins for
+some well-known modules such as express, mongodb, and http, and provide a plugin
+API for third-party developers to create their own.
+
+## Outline
+
+A plugin consists of a set of *patch objects*. A patch object gives information
+about how a file in a module should be patched in order to create trace spans.
+
+Each patch object should contain the following fields:
+
+* `file`: The path to the file whose exports should be patched, relative to the
+  root of the module. You can specify an empty string, or omit this field
+  entirely, to specify the root itself.
```

Comment:
I think `root` is ambiguous here. Maybe program entrypoint or main? (https://docs.npmjs.com/files/package.json#main)